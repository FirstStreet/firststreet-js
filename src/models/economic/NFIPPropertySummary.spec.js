const { isArray } = require('lodash');
const Sut = require('./NfipPropertySummary');

const mock = require('../../__mocks__/nfipPropertySummary.json');

describe('NFIP property summary', () => {
  it('should construct object', () => {
    const result = new Sut(mock);
    expect(result.raw).toMatchSnapshot();
    expect(result.raw.data).toBe(mock.data);
    expect(isArray(result.data)).toBe(true);
    expect(result.data.length).toBe(2);
    expect(result.data[0].constructor.name).toBe('NfipEstimate');
    expect(result.data[1].estimate).toEqual(376);
    expect(result.fsid).toEqual(mock.fsid);
  });

  it('should construct object without child nodes', () => {
    const result = new Sut({});
    expect(result.raw).toMatchSnapshot();
    expect(result.raw.data).toBeUndefined();
    expect(result.data).toBeUndefined();
  });

  it('should construct object with empty child nodes', () => {
    const result = new Sut({ data: [] });
    expect(result.raw).toMatchSnapshot();
    expect(result.raw.data).toEqual([]);
    expect(result.data).toEqual([]);
  });
});
