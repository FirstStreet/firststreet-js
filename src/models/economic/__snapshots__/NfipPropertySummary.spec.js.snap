// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFIP property summary should construct object 1`] = `
Object {
  "data": Array [
    Object {
      "building": 125000,
      "contents": 50000,
      "estimate": 225,
    },
    Object {
      "building": 250000,
      "contents": 125000,
      "estimate": 376,
    },
  ],
  "fsid": 240652458,
}
`;

exports[`NFIP property summary should construct object with empty child nodes 1`] = `
Object {
  "data": Array [],
}
`;

exports[`NFIP property summary should construct object without child nodes 1`] = `Object {}`;
